src/app/app.component.ts

```diff
@@ -1,10 +1,32 @@
 import { Component } from '@angular/core';
+import { mockedCourseList } from './mocks';
+
+interface IData {
+  title: string;
+  description: string;
+  creationDate: Date;
+  duration: number;
+  authors: string[];
+}
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
-  styleUrls: ['./app.component.scss']
+  styleUrls: ['./app.component.scss'],
 })
 export class AppComponent {
   title = 'courses-app';
+
+  data: IData[] = mockedCourseList.map((elem) => ({
+    ...elem,
+    creationDate: new Date(elem.creationDate),
```

Comment:
According to interface creationDate is already a Date

Reply:
The creationDate property of element of mockedCourseList is part of ICourseList interface that considers the property as string, and only after mapping it become Date type